tamr_unify_client/client.py

```diff
@@ -1,62 +1,76 @@
+import logging
+from typing import Optional
 from urllib.parse import urljoin
 
 import requests
-from requests import Response
+import requests.auth
+import requests.exceptions
 
 from tamr_unify_client.dataset.collection import DatasetCollection
 from tamr_unify_client.project.collection import ProjectCollection
 
-# monkey-patch Response.successful
+logger = logging.getLogger(__name__)
 
 
-def successful(self):
-    """Checks that this response did not encounter an HTTP error (i.e. status code indicates success: 2xx, 3xx).
+def successful(response: requests.Response) -> requests.Response:
+    """Checks that this response did not encounter an HTTP error.
```

Comment:
Per recent discussion, it's more clear to say that this will return the current response if no error occurred, or raise the stored error if an error did occur.

Reply:
Agreed, "Check"  could have a boolean connotation. I'll change to "Ensure response does not contain an HTTP error." as the summary description. The "Returns" and "Raises" sections describe the exact behavior.

Reply:
Agreed that phrasing needs to be improved. "Checks" can have a boolean connotation. Changing to "Ensure response does not contain an HTTP error.". The "Returns" and "Raises" sections of this docstring cover the exact behavior.